main.go

```diff
@@ -420,6 +485,41 @@ func postPost(c *gin.Context, d database.IDatabase) {
 	}
 }
 
+func editPost(c *gin.Context, d database.IDatabase, postId string) {
+	var post model.Post
```

Comment:
So for editing a post, I don't think we need to send in a whole post. I think it should just be the new text, and the post Id. WDYT? We really don't need the rest of the info.

Reply:
For example, this is some react crap I fiddled up tonight to start wiring this up:

```
    editPost(text, postId) {
        let baseUrl = config.get('API_ROOT');
        const headers = this.requestHeaders();

        return axios.patch(`${baseUrl}/posts/${postId}`, JSON.stringify({
                Body: body
            }), {
                headers: headers
            }).then(response => {
                return response.data;
            }).catch(error => {
                return error;
            });
    }
```

Given you have the postId in the URI, and the Body as JSON in a patch call, I think you'd have everything you need to update a post. We could potentially send in the UserId of who is doing the editing too, relating to the other comment I made.